### PR TITLE
Module: Add server module

### PR DESCRIPTION
**Pull request by patrickelectric**
Add json and rest servers

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

**Comment by patrickelectric** [comment]
Just opening a review here before creating one in the original repository.

File: MAVProxy/modules/mavproxy_server.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+        # Save status
+        self.status = None
+        self.server = None
+
+        # Start
+        self.start()
+
+    def update_dict(self, mpstate):
+        '''We don't have time to wast'''
```
Comment (jaxxzer) [inline MAVProxy/modules/mavproxy_server.py]: sp. waste

**Comment by jaxxzer** [comment]
<s>This doesn't show up in `module list` output. Neither do depthoutput or gpsinput for that matter.</s> Sorted.

**Comment by patrickelectric** [comment]
Are you running it on source or have you installed it ?
You can use this patch https://github.com/ArduPilot/MAVProxy/pull/504 to run without installing.

**Review by jaxxzer — CHANGES_REQUESTED** [review]
For the jsonserver, Instead of requesting a specific message at a certain interval (this isn't even supported yet in the mavlink api), we should request a specific message, and push the messages out as soon as they are received from the autopilot. This can be done by hooking into (mavlink_packet)[https://github.com/bluerobotics/MAVProxy/blob/master/MAVProxy/modules/mavproxy_DepthOutput.py#L39]

This way updates are only pushed when there is new data, and as soon as it is available.

**Comment by jaxxzer** [comment]
:cry:

**Pull request closed without merging**